apps/explorer/lib/explorer/chain_spec/genesis_data.ex

```diff
@@ -57,53 +72,105 @@ defmodule Explorer.ChainSpec.GenesisData do
   end
 
   @doc """
-  Fetches pre-mined balances and pre-compiled smart-contract bytecodes from genesis.json
+    Fetches and processes the genesis data, which includes pre-mined balances and precompiled smart contract bytecodes.
+
+    This function retrieves the chain specification and precompiled contracts
+    configuration from specified paths in the application settings. Then it
+    asynchronously extends the chain spec with precompiled contracts, imports
+    genesis accounts, and the precompiled contracts' sources and ABIs.
+
+    ## Returns
+    - `Task.t()`: A task handle if the fetch and processing are scheduled successfully.
+    - `:ok`: Indicates no fetch was attempted due to missing configuration paths.
   """
-  @spec fetch_genesis_data() :: Task.t() | :ok
+  @spec fetch_genesis_data() :: Task.t() | :error
   def fetch_genesis_data do
-    path = Application.get_env(:explorer, __MODULE__)[:chain_spec_path]
+    chain_spec_path = get_path(:chain_spec_path)
+    precompiled_config_path = get_path(:precompiled_config_path)
 
-    if path do
+    if is_nil(chain_spec_path) and is_nil(precompiled_config_path) do
+      Logger.warn(fn -> "Genesis data is not fetched. Neither chain spec path or precompiles config path are set." end)
+    else
       json_rpc_named_arguments = Application.fetch_env!(:indexer, :json_rpc_named_arguments)
       variant = Keyword.fetch!(json_rpc_named_arguments, :variant)
 
       Task.Supervisor.async_nolink(Explorer.GenesisDataTaskSupervisor, fn ->
-        case fetch_spec(path) do
-          {:ok, chain_spec} ->
-            case variant do
-              EthereumJSONRPC.Geth ->
-                {:ok, _} = GethImporter.import_genesis_accounts(chain_spec)
-
-              _ ->
-                Importer.import_emission_rewards(chain_spec)
-                {:ok, _} = Importer.import_genesis_accounts(chain_spec)
-            end
-
-          {:error, reason} ->
-            # credo:disable-for-next-line
-            Logger.warn(fn -> "Failed to fetch genesis data. #{inspect(reason)}" end)
-        end
+        chain_spec = fetch_chain_spec(chain_spec_path)
+        precompiles_config = fetch_precompiles_config(precompiled_config_path)
+
+        extended_chain_spec = extend_chain_spec(chain_spec, precompiles_config, variant)
+        import_genesis_accounts(extended_chain_spec, variant)
+
+        import_precompiles_sources_and_abi(precompiles_config)
       end)
+    end
+  end
+
+  @spec get_path(atom()) :: nil | binary()
+  defp get_path(key) do
+    case Application.get_env(:explorer, __MODULE__)[key] do
+      nil -> nil
+      value when is_binary(value) -> value
+    end
```

Comment:
Won't this be the same as just:
```suggestion
    Application.get_env(:explorer, __MODULE__)[key]
```
?

Reply:
Thank you for the suggestion. However, ensuring that the output is consistently either `binary` or `nil` is crucial for the correctness of static analysis later in the code. I found that this consistency helps the static analysis tools to properly handle the configuration value when it is passed as a parameter in subsequent function calls.